### PR TITLE
Address rubocop TODOs

### PR DESCRIPTION
**Pull request by joshuay03**
### Description

Ref: https://github.com/puma/puma/pull/1325

Addresses most of the cops in `.rubocop_todo.yml` using safe autocorrect (`-a`).

Most of those cops were set up to use the default config except for:

- `Layout/SpaceBeforeBlockBraces`
- `Layout/SpaceAroundEqualsInParameterDefault`
- `Layout/SpaceInsideHashLiteralBraces`

which all enforced `no_space`. I've changed these to use the default (`space`) for both consistency with other defaults (e.g. `SpaceAroundOperators`) and (imo) better readability. Let me know if `no_space` is prefered and I can revert back.

The only one I removed from the original TODO config was `Layout/IndentationWidth` which yields multiple inconsistent results when flagging and autocorrecting, even when paired with `Layout/IndentationConsistency` which the rubocop docs recommend. For example, it does this with indented condition assignments:

```ruby
          indented_var = if foo
                           "foo"
          else
            "bar"
          end
```

Also, the only existing cop amended was `Layout/SpaceBeforeBlockBraces`. It now uses the default (`space`) for `EnforcedStyleForEmptyBraces`. My argument is once again consistency, and that the default makes it easier to tell an empty block apart from an empty hash arg when parenthesis are excluded. E.g.

```ruby
# arg
method_call {}

# empty block
method_call { }
```

This also now makes is consistent with the other block cops.

### Your checklist for this pull request

- [x] I have reviewed the [guidelines for contributing](../blob/master/CONTRIBUTING.md) to this repository.
- [ ] I have added (or updated) appropriate tests if this PR fixes a bug or adds a feature.
- [ ] My pull request is 100 lines added/removed or less so that it can be easily reviewed.
- [ ] If this PR doesn't need tests (docs change), I added `[ci skip]` to the title of the PR.
- [ ] If this closes any issues, I have added "Closes `#issue`" to the PR description or my commit messages.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed, including Rubocop.

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -35,7 +39,6 @@ Layout/SpaceAroundKeyword:
   Enabled: true
 
 Layout/SpaceBeforeBlockBraces:
-  EnforcedStyleForEmptyBraces: no_space
```
Comment (joshuay03) [inline .rubocop.yml:38]: This is the default.

File: .rubocop.yml [hunk 2]
```diff
@@ -50,24 +53,57 @@ Layout/TrailingEmptyLines:
...
   Max: 7
 
+# Naming
+
 Naming/ConstantName:
   Enabled: true
 
 Naming/MethodName:
   Enabled: true
-  EnforcedStyle: snake_case
```
Comment (joshuay03) [inline .rubocop.yml:64]: This is already the default.

**Comment by JuanitoFatas** [comment]
It would be great to add the reformatting commit to [.git-blame-ignore-revs](https://docs.github.com/en/repositories/working-with-files/using-files/viewing-a-file#ignore-commits-in-the-blame-view), so they get skipped in git blames 🥇

**Comment by joshuay03** [comment]
> It would be great to add the reformatting commit to [.git-blame-ignore-revs](https://docs.github.com/en/repositories/working-with-files/using-files/viewing-a-file#ignore-commits-in-the-blame-view), so they get skipped in git blames 🥇

Good idea! I'll follow up with this once merged.

**Pull request closed without merging**